### ISSUE TITLE
Trying to train OpenALPR

### ISSUE THREAD

**Issue by RyanLK** [openalpr/openalpr#133]
Hello,

I am trying train openalpr for a new country (Abu Dhabi).

I know I have 2 main things to do:
-Train the OCR, using this guide: https://github.com/openalpr/train-ocr
-Train the detector, using this guide: https://github.com/openalpr/train-detector
First of all, is there any of those 2 that I should do first?

Before seeing this OCR guide, I tried following another one that gave me a result ( a traineddata), but it didn't seem to be working. So as I was searching I found this one, but I am having a hard time following it. 
My problem is whenever I try running classifychars, I keep getting weird errors that I do not understand.
First of all, when I build the project in misc_utilites, all project builds were successful except one (benchmark), but I was still able to get classifychars.exe. So i guessed I could continue. If this is the issue, please let me know.

So I kept going and tried some pictures of plates cropped. It was suggested to have a 250px wide.
Before testing there is something else I need you to clarify me on:
Usage of the classifychar is: openalpr-utils-classifychars.exe country indirectory outdirectory
The "country" is the country I intend to train or a default one already present there?Maybe leu or lus.

Anyway when I tried ad or eu or leu, all gave me the same errors, so again I guessed that wasn't the main issue there.

I tried with a picture that is about 460 px wide:
Result:
(some commands explanation)
Warning: You are running an unsupported version of Tesseract.
Expecting version 3.03, your version is: 3.02
Tesseract couldn't load any languages!
PlatesAD/adplate1.png
OpenCV Error: Assertion failed (s >= 0) in cv::setSize, file C:\builds\2_4_PackS
lave-win32-vc11-shared\opencv\modules\core\src\matrix.cpp, line 116

Then it crashes.

I tried many others with different resolutions, they all keep giving the same error.. So I am not sure whether their size it the issue there or not..

I know its a long post with a lot of questions. I am sorry about that, but I would really like to make it work. Thing is I am not very comfortable with manipulating the program, and I am not finding the guides very simple, so I am running into a lot of issues.

Anything would be really helpful, if what I am doing from the beginning is wrong I would also like to know please.

I hope you guys can help me.
That would be really great.
Thanks in advance.
Ryan

**Comment by matthill** [comment]
I would do the OCR training first.  It's less work than training a detector, and the results may be pretty good without needing a detector.

The pictures used for classifychar should already be cropped at the same aspect ratio as your plate width/height.  Can you upload an example of what image your'e using?

**Comment by RyanLK** [comment]
Hey,

I found out what the issue was. Apparently when I edited the conf file to add the language [ad], it somehow corrupted the file and therefore whenever it was reading from it was getting very weird random numbers every time, so that was what caused the errors.
I think its fixed now. And I added the [ad] language which is a copy paste of the [eu] one just to test now. 

But now when I tried a new error occurred.. This is what happened:
C:\Build\openalpr-master\src\misc_utilities\Release>openalpr-utils-classifychars
.exe ad PlatesAD OutputAD
Usage:
        n               -- Next plate
        p               -- Previous plate
        W               -- Select image and save characters according to OCR res
ults, then go to next image
        s               -- Save characters
        <- and ->       -- Cycle between images
        Ent/space       -- Select plate

Within a plate
        <- and ->               -- Cycle between characters
        [0-9A-Z]                -- Identify a character (saves the image)
        ESC/Ent/Space   -- Back to plate selection
Warning: missing configuration entry for: common->prewarp
Warning: You are running an unsupported version of Tesseract.
Expecting version 3.03, your version is: 3.02
PlatesAD/adplate0.png
OpenCV Error: Assertion failed (scn == 3 || scn == 4) in cv::cvtColor, file C:\b
uilds\2_4_PackSlave-win32-vc11-shared\opencv\modules\imgproc\src\color.cpp, line
 3737

It opened a small window called "Original", empty with a grey background. Then it crashed with the above error.

Any idea what this error means?

This is the example that caused the above error.
![adplate0](https://cloud.githubusercontent.com/assets/12894842/8204156/2e2a4806-14ef-11e5-9c3a-964ea920be40.png)
